### PR TITLE
Jr/tw baseof

### PR DESCRIPTION
**Pull request by jeffrummel**
Converts Main.html to Tailwind

https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/tips/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/contact/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/about/jobs/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/authors/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/contact/thanks/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/newsletters/thanks/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/newsletters/sorry/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/search/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/newsletters/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/embeds/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/donate/thanks/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/donate/sorry/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/404/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/contact/thanks/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/about/terms/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/about/diversity/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/about/mission/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/about/faq/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/about/privacy/
https://deploy-preview-368--poor-richard-spotlightpa.netlify.app/about/partners/

### CODE REVIEW

**Comment by jeffrummel** [comment]
@carlmjohnson I have a lot of Q/A to perform but will let you know when it's ready for review

**Comment by carlmjohnson** [comment]
One trick for the QA: You can go to master branch, run `hugo`, duplicate the public folder, come back to this branch, run `hugo` again, and then drop both folders on /Applications/Xcode.app/Contents/Applications/FileMerge.app. FileMerge can compare two directories and tell you which files in them are different. Then you would know if anything change that shouldn't have changed.

**Comment by jeffrummel** [comment]
@carlmjohnson I've updated Newsletters page to align the checkbox to the baseline in Safari

File: static/img/icons/check.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc. --><path d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"/></svg>
```
Comment (carlmjohnson) [inline static/img/icons/check.svg:1]: We're already importing check.svg from fontawesome through NPM. It's "check-svg" "@fontawesome/solid/check.svg".

**Review by carlmjohnson — COMMENTED** [review]
Other stuff:

- [ ] All of the input boxes need more padding on the left. We need more consistency around the focus and hover styles. In general, our input boxes are very inconsistent because Tailwind doesn't make it easy to have consistent input boxes. I think it's time for us to break down and create a proper input box house style. Maybe as a rule in tw.css or maybe as a partial. I'm leaning towards tw.css because the partial would need to be able to accommodate one off rules somehow, and it's probably easier to do as a rule, like `<input class="spl-input text-xl">` vs. `{{ partial "tw/input.html" (dict ...) }}`.
- [ ] The privacy policy is broken on mobile. It's also broken-ish on the existing page because of the table, but the table is in a special box that makes it scroll horizontally.

File: layouts/partials/tw/thumb-h-item.html [hunk 2]
```diff
@@ -79,14 +79,13 @@
...
           >
-                      <a
+            <a
               href="{{ .link }}"
-              class="group-hover:underline text-s-7"
+              class="text-s-7 group-hover:underline"
               data-ga-category="byline"
             >
-            {{ .blurb }}
+              {{ .blurb }}
```
[the remark this answers is not in the document]
Comment (jeffrummel) [reply layouts/partials/tw/thumb-h-item.html:87]: prettier wants to keep the tab there

File: layouts/partials/tw/message-block.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    class="article-content mx-auto max-w-prose overflow-hidden rounded bg-white shadow"
+  >
+    <div class=" py-3 px-5 text-white" style="background: rgb({{ .color }})">
+      <h3 class="text-2xl font-extrabold">{{ .title }}</h3>
+      {{ with .subhed }}
+        <p>{{ . }}</p>
+      {{ end }}
+    </div>
+    <div
+      class="py-3 px-5 text-base font-light"
```
[the remark this answers is not in the document]
Comment (jeffrummel) [reply layouts/partials/tw/message-block.html]: Updated: https://github.com/spotlightpa/poor-richard/pull/368/commits/790aaa40caf8000598ae7039e2f5ec1151479c61#diff-4011f16c257a49b7914b408447011f543b193706c0d0b968c3c1390713e4f0cd

File: layouts/partials/tw/message-block.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      style="background: rgba({{ .color }},.1);color: rgb({{ .color }}; "
```
[the remark this answers is not in the document]
Comment (jeffrummel) [reply layouts/partials/tw/message-block.html]: updated: https://github.com/spotlightpa/poor-richard/pull/368/commits/790aaa40caf8000598ae7039e2f5ec1151479c61

File: content/search.md [hunk 5]
```diff
@@ -9,4 +9,5 @@ suppress-title = true
 
 Search Spotlight PA with the box below:
 
-{{<about/news-search>}}
+
+{{<news-search>}}
```
[the remark this answers is not in the document]
Comment (jeffrummel) [reply content/search.md:13]: Updated: https://github.com/spotlightpa/poor-richard/pull/368/commits/6ff18f8f2cdaaf082b2073865cae7bdcbd22c734

File: content/about/about-jobs.md [hunk 6]
```diff
@@ -8,7 +8,7 @@ aliases = ["/jobs/"]
 +++
 ### Current openings:
 
-* [Membership Manager](#membership-manager)
+* Membership Manager](#membership-manager)
```
Comment (carlmjohnson) [inline content/about/about-jobs.md:11]: This looks like a typo. Is the branch up to date?

Comment (jeffrummel) [reply content/about/about-jobs.md]: That is a typo. I second guessed myself after merging with master. Fixed with the latest commit: https://github.com/spotlightpa/poor-richard/pull/368/commits/67898a9a3c0df133604829146f125bf994de406f

**Comment by jeffrummel** [comment]
>     * [ ]  The privacy policy is broken on mobile. It's also broken-ish on the existing page because of the table, but the table is in a special box that makes it scroll horizontally.

The table in the pp is fixed-ish. I'm still not happy with the customizations allowed by tw in a markdown file. These basic pages are loading `.content` into the article container so we could create a table style in tw.css to be applied inside articles or even a table shortcode.

File: assets/css/tw.css [hunk 7]
```diff
@@ -183,6 +183,12 @@
 .content-aside ol > li + li {
   @apply mt-3;
 }
+.spl-input {
+  @apply w-full rounded border border-s-4 py-2 pl-4 pr-7 font-sans outline-none ring-2 ring-transparent focus:ring-yellow;
+}
```
Comment (carlmjohnson) [inline assets/css/tw.css:188]: At one point I wrote a tool to go through our CSS and see how many times a set of classes were used together: https://github.com/carlmjohnson/classy

Here's the output from master:

<details>

```
18  "fill-current h-4 w-4"
12  "absolute inline-flex inset-y-0 items-center justify-center leading-none left-0 pointer-events-none text-s-4 w-10 z-10"
11  "max-w-prose mx-auto w-full"
 8  "h-4 mr-2 w-4"
 7  "bg-white px-6 py-8"
 7  "has-colored-svg has-text-white icon"
 7  "flex-grow relative"
 6  "max-w-screen-lg mx-auto"
 6  "bg-red font-semibold inline-block px-4 py-3 rounded-br sm:text-lg text-white tracking-wider uppercase"
 6  "control is-expanded"
 6  "appearance-none bg-white block border-2 border-s-4 focus:ring-2 outline-none placeholder-opacity-100 placeholder-s-4 px-4 py-2 ring-red-7 rounded text-s-8 tracking-wider w-full"
 6  "fill-current h-6 w-6"
 6  "items-stretch mt-8 mx-auto sm:flex sm:max-w-2xl"
 5  "bg-white md:pb-6 pb-2 pt-12 px-4"
 5  "block font-extrabold mt-4"
 5  "bg-white p-6"
 5  "image is-64x64"
 5  "max-w-6xl mx-auto px-5"
 5  "fill-current h-8 inline-block w-48"
 4  "absolute bg-s-4 inset-0 opacity-75"
 4  "border border-red font-raleway font-semibold leading-tight rounded shadow"
 4  "border-b-4 border-black font-extrabold font-sans leading-none pb-2 text-xl tracking-wider uppercase"
 4  "flex flex-0 items-center mr-2"
 4  "basis-0 flex gap-2 items-start justify-start"
 4  "flex items-center"
 4  "absolute backdrop-blur-sm backdrop-filter inset-0"
 4  "bg-yellow-4 border-2 border-yellow-5 focus:ring-2 font-bold outline-none px-8 py-2 ring-yellow-6 rounded text-black tracking-widest uppercase w-full"
 4  "flex-grow hover:underline text-s-6"
 4  "mt-1 text-left"
 4  "appearance-none bg-white block border-2 border-s-4 focus:ring-2 outline-none pl-10 placeholder-opacity-100 placeholder-s-4 py-2 ring-yellow-6 rounded text-s-8 tracking-wider w-full"
 4  "mt-2 text-left"
 4  "hidden sm:block"
 4  "mb-4 md:mt-4 mr-3 mt-10 text-right text-s-9"
 4  "mt-4 sm:ml-1 sm:mt-0"
 4  "max-w-screen-lg mx-auto w-full"
 4  "max-w-screen-md mx-auto"
 4  "break-all text-blue"
 3  "flex gap-2 items-baseline"
 3  "flex flex-row items-center md:my-0 mr-10 my-2"
 3  "flex justify-center mt-12"
 3  "fixed inset-0 z-20"
 3  "is-monochrome is-rounded"
 3  "fill-current h-4 inline-block w-4"
 3  "appearance-none border-s-1 checked:bg-black h-6 mr-4 order-first rounded-full shadow-[0_0_0_3px_rgba(255,255,255,1)] w-6"
 3  "leading-tight min-w-full mt-4 sm:text-base sm:w-0 space-y-4 text-sm"
 3  "duration-500 hover:text-yellow text-navy transition-colors underline"
 3  "duration-500 hover:text-yellow text-navy transition-colors"
 3  "mt-12 text-center"
 3  "duration-500 hover:text-yellow text-blue transition-colors"
 3  "drop-shadow-outline fill-current filter h-8 hover:text-yellow lg:h-10 text-white"
 3  "is-child tile"
 3  "font-bold font-sans leading-none md:text-6xl sm:text-4xl text-3xl text-center tracking-wide uppercase"
 3  "bg-white font-serif pb-12 pt-6 px-4"
 3  "control has-text-left"
 3  "control has-icons-left has-icons-right is-expanded"
 3  "accent-black border border-black h-4 hover:accent-yellow-4 mr-3 p-1 rounded w-4"
 3  "is-4 subtitle"
 3  "columns is-centered is-multiline"
 3  "font-bold font-sans uppercase"
 3  "column is-three-fifths"
 3  "font-bold text-lg uppercase"
 3  "absolute flex inset-1 overflow-y-auto"
 3  "absolute bottom-0 inset-x-0 sm:hidden"
 3  "font-normal font-sans leading-tight my-3 text-base text-byline tracking-normal"
 3  "column is-3"
 3  "max-w-screen-lg mb-8 mt-4 mx-auto w-full"
 3  "font-sans mb-4 sm:mb-0 sm:mt-[95px] sm:shape-outside sm:w-auto w-full"
 3  "border-b-2 border-black tracking-wide whitespace-nowrap"
 3  "icon is-right validator-icon"
 3  "icon is-large is-left"
 3  "group-hover:underline text-s-7"
 3  "max-w-screen-lg mx-auto py-12"
 3  "has-text-weight-semibold is-uppercase"
 3  "button has-text-weight-semibold is-danger"
 3  "absolute bg-gradient-to-t from-white inset-0 sm:hidden"
 3  "flex gap-1 items-center"
 3  "bg-red-6 hover:bg-red-5 inline-block px-4 py-3 rounded text-white"
```

</details>

File: layouts/404.html [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {{ define "main" }}
   {{ $content :=  "The requested page was not found. <a href='/' class='has-text-weight-bold'>Go home?</a>" | safeHTML }}
-  {{ $dict := dict "class" "is-danger" "title" "Not Found" "subtitle" "Error 404" "content" $content }}
-  {{ partial "component/message-block.html" $dict }}
+  {{ $dict := dict "color" "215 0 0" "title" "Not Found" "subtitle" "Error 404" "content" $content }}
```
Comment (carlmjohnson) [inline layouts/404.html:3]: Old versions of Hugo didn't allow multiline dictionaries. Go ahead break this one up to make it readable:

```
{{ $dict := dict 
  "color" "215 0 0" 
  "title" "Not Found"
  "subtitle" "Error 404" 
  "content" $content 
}}
```

File: layouts/_default/main.html [hunk 9]
```diff
@@ -4,122 +4,51 @@
...
     </div>
-  </article>
+  {{ end }}
+
+  {{ range $i, $blocks := .Params.blocks }}
+    {{ $id := urlize .title }}
+    <div
+      id="{{ $id }}"
+      x-data
+      x-intersect.half="$dispatch('intersect:support', $el.id)"
```
Comment (carlmjohnson) [inline layouts/_default/main.html:42]: I think this is for the sticky menu on the about page. Do we need it elsewhere?

File: assets/css/tw.css [hunk 10]
```diff
@@ -183,6 +183,12 @@
...
+.spl-select {
+  @apply block w-full appearance-none rounded border border-s-4 bg-white py-2 pl-4 pr-7 font-sans text-s-6 shadow outline-none ring-2 ring-transparent focus:ring-yellow active:ring-yellow-6;
```
Comment (carlmjohnson) [inline assets/css/tw.css:190]: Select has a shadow but input doesn't. Either way is fine, but they should be the same.

**Review by carlmjohnson — COMMENTED** [review]
Looks good. Needs `yarn format`.

File: layouts/404.html (hunk 8 above)
Comment (carlmjohnson) [reply layouts/404.html:3]: Also the other ones like this.

File: layouts/_default/main.html (hunk 9 above)
Comment (jeffrummel) [reply layouts/_default/main.html]: I've removed this: https://github.com/spotlightpa/poor-richard/pull/368/commits/08f696ce271a7809053fd11f3d6ebd26b8fb11b4

File: layouts/_default/author.terms.html [hunk 11]
```diff
@@ -1,60 +1,82 @@
...
+        {{ range $pg.Pages.ByTitle }}
+          {{ $author := partial "helper/get-author" .Title }}
+          <li
+            id="{{ $author.slug }}"
+            class="flex items-center justify-start space-x-4 mb-4"
+          >
+            {{ if $author.headshot }}
+              {{ $sizes := (slice
+                (dict "width" 64 "height" 64)
+                (dict "width" 64 "height" 64)
```
Comment (carlmjohnson) [inline layouts/_default/author.terms.html:19]: I think this was supposed to be 64x64 then 128x128. It's blurry on my retina MacBook:

image

**Review by carlmjohnson — APPROVED** [review]

**Pull request merged**